controllers/cfapi/controller.go

```diff
@@ -412,3 +385,76 @@ func (r *Reconciler) uninstall(ctx context.Context, config v1alpha1.Installation
 		State: installable.ResultStateSuccess,
 	}, nil
 }
+
+type gateway struct {
+	k8sClient  client.Client
+	kymaClient *kyma.Client
+	Type       string
+	Service    *corev1.Service
+}
+
+func newGateway(k8sClient client.Client, kymaClient *kyma.Client, gatewayType string) gateway {
+	gw := gateway{
+		k8sClient:  k8sClient,
+		kymaClient: kymaClient,
+		Type:       v1alpha1.GatewayTypeContour,
+	}
+	if gatewayType != "" {
+		gw.Type = gatewayType
+	}
+
+	return gw
+}
+
+func (g *gateway) Validate(ctx context.Context) error {
+	if g.Type != v1alpha1.GatewayTypeContour && g.Type != v1alpha1.GatewayTypeIstio {
```

Comment:
Would it not be easier to validate accepted values of the `gatewayType` via a kubebuilder annotation?